src/app_service/service/message/service.nim

```diff
@@ -526,8 +526,8 @@ QtObject:
       self.finishAsyncSearchMessagesWithError(chatId, "search messages response doesn't contain messages array")
       return
 
-    if (messagesArray.kind != JArray):
-      self.finishAsyncSearchMessagesWithError(chatId, "expected messages json array is not of JArray type")
+    if (messagesArray.kind != JArray and messagesArray.kind != JNull):
```

Comment:
Should this be ` or messagesArray.kind == JNull` instead? 
Actually, I'm not sure why the original check didn't work, if the return was null, the `JArray` check should have caught it?

Reply:
@micieslak you can also handle this in `async_task.nim`, task `asyncSearchMessagesInChatTask`, if result is null, you can set there `"messages":[]"`, but definitely condition `if (messagesArray.kind != JArray)` proceeds further only if `messagesArray` is array, so `and messagesArray.kind != JNull` is redundant (if something is `JNull` it is not `JArrray`).

Reply:
Originally, when the type was different then JArray, finish...WithError was called. So it happened for JNull as well, which imo should be treated as a valid kind when we have no messages. Now both JArray and JNull are considered valid kinds.

> but definitely condition if (messagesArray.kind != JArray) proceeds further only if messagesArray is array, so and messagesArray.kind != JNull is redundant (if something is JNull it is not JArrray)

We have != so it proceeds only if messagesArray is NOT JArray, right?

I can fix it in "async_task.nim" as you suggested as well, I initially fixed it there but then moved here :)

Reply:
The problems with fixing it in `asyncSearchMessagesInChatTask` are as follows:
- we would have to assume that response obtained from go backend contains `"messages"` field and replace with [] if it's null. But only in `onAsyncSearchMessages` we validate if this field exists.
- (minor) the same check should be repeated in `asyncSearchMessagesInChatsAndCommunitiesTask`

So I would prefer to keep the response unmodified in `asyncSearchMessagesInChatTask` and `asyncSearchMessagesInChatsAndCommunitiesTask` and validate exclusively in `onAsyncSearchMessages`.

I tweaked the condition a bit for better readability.

@saledjenic, @jrainville, what do you think?

Reply:
@saledjenic  `if response.result.kind != JNull` will be always true. We would have to check if `response["messages"]` exists and then replace with an empty array if null, what becomes slightly more complicated and needs to be repeated in `asyncSearchMessagesInChatsAndCommunitiesTask`. So I would opt for current solution which seems to be simpler, I hope.